test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,356 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Functional\Specification;
+
+use OnMoon\OpenApiServerBundle\Specification\Definitions\Specification;
+use OnMoon\OpenApiServerBundle\Specification\Definitions\SpecificationConfig;
+use OnMoon\OpenApiServerBundle\Specification\SpecificationLoader;
+use OnMoon\OpenApiServerBundle\Specification\SpecificationParser;
+use OnMoon\OpenApiServerBundle\Types\ScalarTypesResolver;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\Config\FileLocatorInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+use Symfony\Contracts\Cache\TagAwareCacheInterface;
+use Throwable;
+
+use function array_pop;
+use function Safe\sprintf;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\Specification\SpecificationLoader
+ */
+class SpecificationLoaderTest extends TestCase
+{
+    /** @var mixed $specificationParser */
+    private $specificationParser;
+    /** @var mixed $fileLocator  */
+    private $fileLocator;
+    /** @var mixed $cache  */
```

Comment:
Why mixed? This will work with typed properties, mocks are still subtypes of the mocked classes.